### PR TITLE
React redux capstone project

### PR DESCRIPTION
**Pull request by yasinabdmahmood**
### Project features

**API**
[REST COUNTRIES API](https://restcountries.com/) was chosen for this project which provides information about countries of the world.

**Design**
-  These [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)) were followed , including:
  - Colors (select one main color for your website).
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.

**Interactions**
- Home page
  - When the page loads, the webapp shows the list of countries that could be filtered out by continent
  - Along with the country name,  some numeric values per country are displayed that come from the API.
  - When the user clicks (or taps) on a country item, the application navigates to the details page.

- Details page
  - In the details page, the webapp retrieves data from the API to show detailed data in the country.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

**Testing implementation**
- Created unit tests for pure functions in the Redux code.
- Created integration tests for the application using the React Testing Library.

**Technical implementation**

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should is stored in the Redux store.
- The data retrieved from the API is filtered using a Filter stateless component.
- Every page (the main page and the pages for each item) has a unique route within the SPA.
- The project is deployed and accessible online (the link to the online version of the app can be found in README file).

### CODE REVIEW

File: src/components/Home/Home.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          <option value="Europe">Europe</option>
+          <option value="Oceania">Oceania</option>
+        </select>
+      </div>
+      <CountryList countries={filter || countries} />
+    </div>
+  );
+}
+
+export default Home;
```
Comment (codecaiine) [inline src/components/Home/Home.js:44]: - [ ] Good job so far with your project it looks great and works as expected :clap:  But we have an issue with your design :cry: As you check on the project requirement, we expect you to work on mobile design and we can see some differences between your design and the original design. Please try to make your design similar to the original design for both the home and details page as a developer it is important to pay attention to details :+1: 

#### Figma design
metrics_app

#### You design
Screenshot from 2022-10-11 23-35-47
Screenshot from 2022-10-11 23-34-13

- [ ] Please make sure to follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), especially the layout: composition and space between elements as request for this project :+1:

File: README.md [hunk 2]
```diff
@@ -1,70 +1,84 @@
...
-### `npm run build`
+- live demo: [Click here](https://tubular-cupcake-f62e8d.netlify.app)
 
-Builds the app for production to the `build` folder.\
-It correctly bundles React in production mode and optimizes the build for the best performance.
+### Video Link
 
-The build is minified and the filenames include the hashes.\
-Your app is ready to be deployed!
+- [Click here](https://drive.google.com/file/d/1VvIjeALtPvCYoudo-fFKRO6J01mEHtho/view?usp=sharing)
```
Comment (codecaiine) [inline README.md:28]: - [ ] Your demo video is great but please make sure that your presentation video contains a highlight about a piece of the code the student finds interesting or is proud of making. :smile: 

- [ ] I noticed that your readme doesn't have a link to issue , please add it , that will help other contribute to your project and will make your readme more professional :smile:

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @yasinabdmahmood , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 2 above)
Comment (yasinabdmahmood) [reply README.md:28]: Hello @codecaiine  , I have already demonstrated an extra feature in my video which is the scroll-up icon I created to automatically scroll up to the top of the page.

**Review by Bondok6 — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

## Hi @yasinabdmahmood 🙋‍♂️, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional ✅
- Your repo is professional ✅
- Well structured files ✅
- Professional README.md file 💯
- No Linter errors ✅


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**